lib/tilex_web/controllers/auth_controller.ex

```diff
@@ -32,21 +32,36 @@ defmodule TilexWeb.AuthController do
     |> redirect(to: "/")
   end
 
-  defp authenticate(%{info: info}) do
-    email = Map.get(info, :email)
-    name = Developer.format_username(Map.get(info, :name))
-
-    case String.match?(email, ~r/@#{Application.get_env(:tilex, :hosted_domain)}$/) do
-      true ->
+  defp authenticate(%{info: %{email: email, name: name}}) when is_binary(name) do
+    case authorized(email) do
+      {:ok, email} ->
         attrs = %{
           email: email,
-          username: name
```

Comment:
This could fail previously if name wasn't present in info or wasn't a binary.

Reply:
See [here](https://developers.google.com/identity/protocols/OpenIDConnect#obtainuserinfo)